examples/oauth/app/routes/oauth.callback.tsx

```diff
@@ -0,0 +1,31 @@
+import { useEffect } from 'react'
+import type { ActionFunction } from 'remix'
+import { useSubmit } from 'remix'
+import { authenticator } from '~/auth.server'
+import { supabaseClient } from '~/supabase.client'
+
+export const action: ActionFunction = async({ request }) => {
+  await authenticator.authenticate('sb-oauth', request, {
+    successRedirect: '/private',
+    failureRedirect: '/login',
+  })
+}
+
+export default function OAuth() {
+  const submit = useSubmit()
+
+  useEffect(() => {
+    supabaseClient.auth.onAuthStateChange((event, session) => {
+      if (event === 'SIGNED_IN') {
+        const formData = new FormData()
+        formData.append('session', JSON.stringify(session))
+
+        submit(formData, { method: 'post' })
+        supabaseClient.auth.signOut()
```

Comment:
I believe this will invalidate the refresh token. You could consider removing just the cookie? btw Great work, I used some of it for Supabase Google Oauth

Reply:
supabaseClient.auth.signOut() will only remove supabase local storage instance ;)  (we only use session cookie with remix-auth :D )

Reply:
I set my supabase token lifetime to 10s (for testing), and remix-auth-supabase always handle the session refresh gracefully :D

Reply:
Thanks for testing. It seems there was something wrong with the specific user in supabase, I received "invalid refresh token" on each refresh. I've tried with a new user and there were no more issues 🤔 👍

Reply:
I try all the nigth (and Github micro ban me haha) and didn't spot any issue.
I have to admit that I don't use my example in production (only email password). That was a challenge 😅

Someone open an issue for user without verified email, maybe a clue ? (Its issue was on classic email password authent)